src/System.CommandLine/Parsing/SymbolResult.cs

```diff
@@ -19,6 +20,11 @@ public abstract class SymbolResult
             Symbol = symbol ?? throw new ArgumentNullException(nameof(symbol));
 
             Parent = parent;
+
+            if (parent?.Root is {} root)
+            {
+                _root = root;
+            }
```

Comment:
Is thee any reason we can't directly assign rather than do the null check?
```suggestion
            _root = parent?.Root;
```